File: bot-api/java/src/main/java/dev/robocode/tankroyale/botapi/internal/json/JsonConverter.java
package dev.robocode.tankroyale.botapi.internal.json;

import com.google.gson.JsonObject;

public final class JsonConverter {

    public static String toJson(Object obj) {
        return GsonFactory.getGson().toJson(obj);
    }

    public static <T> T fromJson(String json, Class<T> type) {
        return GsonFactory.getGson().fromJson(json, type);
    }

    public static <T> T fromJson(JsonObject jsonObject, Class<T> type) {
        return GsonFactory.getGson().fromJson(jsonObject, type);
    }
}

File: bot-api/dotnet/api/src/internal/json/JsonConverter.cs
using System;
using Newtonsoft.Json;

namespace Robocode.TankRoyale.BotApi.Internal.Json;

public static class JsonConverter
{
    private static readonly JsonSerializerSettings Settings;

    static JsonConverter()
    {
        Settings = new JsonSerializerSettings();
        Settings.Converters.Add(new ColorJsonConverter());
    }

    public static string ToJson(object obj)
    {
        return JsonConvert.SerializeObject(obj, Settings);
    }

    public static T FromJson<T>(string json) where T : class
    {
        return (T)JsonConvert.DeserializeObject(json, typeof(T), Settings);
    }

    public static object FromJson(string json, Type type)
    {
        return JsonConvert.DeserializeObject(json, type, Settings);
    }
}
